projects/plugins/boost/app/lib/critical-css/Critical_CSS_State.php

```diff
@@ -383,10 +392,21 @@ private function collate_column( $column ) {
 	/**
 	 * Get the provider urls.
 	 *
+	 * @param bool $with_provider_keys Indicate whether to include generation query args
 	 * @return array
 	 */
-	public function get_provider_urls() {
-		return $this->collate_column( 'urls' );
+	public function get_provider_urls( $with_provider_keys = false ) {
```

Comment:
Passing a flag to a method as an argument is a good sign that the function is breaking SRP.